make-distribution.sh

```diff
@@ -40,57 +40,25 @@ TERM=xterm-color ./dev/make-distribution.sh \
     ${HIVE_INSTALL_FLAG:+"-Phive"} \
     -DskipTests
 
-SPARK_MAJOR_VERSION="$(echo "${SPARK_VERSION}" | cut -d '.' -f1)"
-HADOOP_MAJOR_VERSION="$(echo "${HADOOP_VERSION}" | cut -d '.' -f1)"
-HIVE_HADOOP3_HIVE_EXEC_URL=${HIVE_HADOOP3_HIVE_EXEC_URL:-https://github.com/guangie88/hive-exec-jar/releases/download/1.2.1.spark2-hadoop3/hive-exec-1.2.1.spark2.jar}
-
-# Replace Hive for Hadoop 3 since Hive 1.2.1 does not officially support Hadoop 3 when using Spark 2.y.z
-# Note docker-image-tool.sh takes the jars from assembly/target/scala-2.*/jars
-if [[ "${WITH_HIVE}" = "true" ]] && [[ "${SPARK_MAJOR_VERSION}" -eq 2 ]] && [[ "${HADOOP_MAJOR_VERSION}" -eq 3 ]]; then
-    HIVE_EXEC_JAR_NAME="hive-exec-1.2.1.spark2.jar"
-    TARGET_JAR_PATH="$(find assembly -type f -name "${HIVE_EXEC_JAR_NAME}")"
-    curl -LO "${HIVE_HADOOP3_HIVE_EXEC_URL}" && mv "${HIVE_EXEC_JAR_NAME}" "${TARGET_JAR_PATH}"
-    # Spark <= 2.4 uses ${TARGET_JAR_PATH} for Docker COPY, but Spark >= 3 uses dist/jars/
-    cp "${TARGET_JAR_PATH}" "dist/jars/"
-fi
-
 SPARK_MAJOR_VERSION="$(echo "${SPARK_VERSION}" | cut -d '.' -f1)"
 SPARK_MINOR_VERSION="$(echo "${SPARK_VERSION}" | cut -d '.' -f2)"
+HADOOP_MAJOR_VERSION="$(echo "${HADOOP_VERSION}" | cut -d '.' -f1)"
 
-if [[ ${SPARK_MAJOR_VERSION} -eq 2 && ${SPARK_MINOR_VERSION} -eq 4 ]]; then  # 2.4.z
-    # Same Dockerfiles as Spark v2.4.8, but allow override of base image to use Debian Buster
-    # and not using PYTHONENV and instead copies pyspark out like Spark 3.y.z
-    DOCKERFILE_BASE="../overrides/base/2.4.z/Dockerfile"
-    DOCKERFILE_PY="../overrides/python/2.4.z/Dockerfile"
-else
-    DOCKERFILE_BASE="./resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile"
-    DOCKERFILE_PY="./resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile"
-fi
+DOCKERFILE_BASE="./resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile"
+DOCKERFILE_PY="./resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile"
 
 if [[ ${SPARK_MAJOR_VERSION} -eq 3 && ${SPARK_MINOR_VERSION} -ge 4 ]]; then  # >=3.4
     # From Spark v3.4.0 onwards, openjdk is not the prefered base image source as it i
     # deprecated and taken over by eclipse-temurin. slim-buster variants are not available
     # on eclipse-temurin at the moment.
-    IMAGE_VARIANT="jre"
```

Comment:
`jre-8` no longer works properly with the current setup as it is upgraded to Ubuntu 22, and the python within that version raises an error when user install python packages globally.  However, [Dockerfile](https://github.com/apache/spark/blob/v3.5.1/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile) packaged in Spark still installs packages globally ([see prev GH action logs here](https://[productionresultssa8.blob.core.windows.net/actions-results/b4190c58-dba2-4ae0-b4a2-b51cef9ddb45/workflow-job-run-35937173-daf7-58ec-4575-8903d4c5ba0f/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-09T08%3A35%3A57Z&sig=itpVNpeTGqLIADqA1bXp8Wj%2BXkA8M3BUVaHH0hei8R4%3D&ske=2024-09-09T17%3A57%3A31Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-09T05%3A57%3A31Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-09-09T08%3A25%3A52Z&sv=2024-05-04](https://productionresultssa8.blob.core.windows.net/actions-results/b4190c58-dba2-4ae0-b4a2-b51cef9ddb45/workflow-job-run-35937173-daf7-58ec-4575-8903d4c5ba0f/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-09T08%3A35%3A57Z&sig=itpVNpeTGqLIADqA1bXp8Wj%2BXkA8M3BUVaHH0hei8R4%3D&ske=2024-09-09T17%3A57%3A31Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-09T05%3A57%3A31Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-09-09T08%3A25%3A52Z&sv=2024-05-04))

![image](https://github.com/user-attachments/assets/ef6c5fa3-b73d-49c9-87a6-290869860abd)